atom/browser/native_window_mac.mm

```diff
@@ -95,6 +95,33 @@ - (void)windowDidResignMain:(NSNotification*)notification {
   shell_->NotifyWindowBlur();
 }
 
+- (NSSize)windowWillResize:(NSWindow *)sender toSize:(NSSize)frameSize {
+  NSSize newSize = frameSize;
+  double interiorContentAspectRatio = shell_->GetInteriorContentAspectRatio();
+
+  if (interiorContentAspectRatio > 0.0) {
+    NSRect windowFrame = [sender frame];
+    NSRect contentViewBounds = [[sender contentView] bounds];
```

Comment:
The code only computes the difference between the content size and the windows size. That is necessary for allowing the code to maintain the aspect ratio within the window while accounting for the title bar on Mac OS X since there are several different sizes of title controls. Is there a scenario where you're drawing outside of the contentView's bounds? As far as I can tell there shouldn't be a difference between the two. We can certainly change the implementation to your suggested one for the time being. Is all of the frameless window implementation within native_window_mac.mm?